client/signup/config/steps.js

```diff
@@ -87,5 +88,17 @@ module.exports = {
 		},
 		dependencies: [ 'siteSlug' ],
 		providesDependencies: [ 'theme', 'images' ]
+	},
+
+	'jetpack-user': {
+		stepName: 'jetpack-user',
+		apiRequestFunction: stepActions.createAccount,
+		providesToken: true,
+		props: {
+			headerText: i18n.translate( 'Create an account for Jetpack' ),
+			subHeaderText: i18n.translate( 'You\'re moments away from connecting Jetpack.' )
+		},
+		dependencies: [ 'jetpackRedirect' ],
+		providesDependencies: [ 'bearer_token', 'username', 'jetpackRedirect' ]
```

Comment:
… which isn't to say that this is the only way to pass that data to `createAccount`, but it won't appear in `dependencies` unless it is provided and required by the same step.

You can avoid adding `jetpackRedirect` to `dependencies` / `providedDependencies` entirely by adding it to object that is the first argument of `submitSignupStep`, because any data included in the first parameter of `submitSignupStep` is also passed to the action. `flowName` is an example of this – notice how it is added to the request in `signup_flow_name` in `createAction`.

Let me know if that doesn't make sense.